### ISSUE TITLE
Multifolder workspace support for VS Code

### ISSUE THREAD

**Issue by cgranade** [microsoft/qsharp-compiler#93]
**Is your feature request related to a problem? Please describe.**
The current version of the VS Code extension is optimized for single-folder workspaces, and in several places assumes that the active workspace contains exactly one folder. This makes it difficult to use the extension in some scenarios, such as that discussed in microsoft/quantumkatas#150.

**Describe the solution you'd like**
The lifecycle management for the language server should be extracted from the extension lifecycle for the VS Code extension, making it easier to spin up and spin down copies of the language server process as folders are added to and removed from the active workspace.

**Comment by bettinaheim** [comment]
@cgranade I happened to see a branch with some commits that look like you might be working on this? If so, could you please indicate that here to avoid confusion, especially since it's marked as help-wanted. Could you please comment with a link to the branch where you are working, and occasional status updates in case people want to join would be great as well!

**Comment by cgranade** [comment]
Good point, @bettinaheim. I've been working on the rather badly misnamed [`cgranade/vscode-launch-jupyter`](https://github.com/microsoft/qsharp-compiler/tree/cgranade/vscode-launch-jupyter) branch, as I was refactoring to make it easier to add in new calls out to .NET Global Tools and got a bit carried away. So far, I'm working on trying to find a bug with how the language client opens up the TCP/IP port to talk to the language server, such that it's not quite ready for prime time. Definitely happy if someone wants to pick up from my experimentation!
